src/libponyc/type/cap.c

```diff
@@ -86,29 +70,25 @@ bool is_cap_sub_cap(token_id sub, token_id subalias, token_id super,
       return false;
   }
 
-  if((sub == super) || (super == TK_TAG) || (super == TK_CAP_ANY_BIND))
+  if((sub == super) || (super == TK_TAG))
     return true;
 
-  // Every possible value of sub must be a subtype of every possible value of
-  // super.
+  // Every possible instantiation of sub must be a subtype of every possible
+  // instantiation of super.
   switch(sub)
   {
     case TK_ISO:
       switch(super)
       {
-        case TK_ISO:
         case TK_TRN:
         case TK_REF:
         case TK_VAL:
         case TK_BOX:
-        case TK_CAP_READ:
-        case TK_CAP_SHARE:
-        case TK_CAP_SEND:
-        case TK_CAP_ALIAS:
-        case TK_CAP_ANY:
-
-        case TK_ISO_BIND:
-        case TK_CAP_SEND_BIND:
+        case TK_CAP_READ:  // {iso, trn, ref, val, box}
+        case TK_CAP_SHARE: // {val, tag}
+        case TK_CAP_SEND:  // {iso, val, tag}
+        case TK_CAP_ALIAS: // {ref, val, box, tag}
+        case TK_CAP_ANY:   // {iso, trn, ref, val, box, tag}
```

Comment:
As you can see, I've added comments for all the gencaps to show their expansion inline.  I found this really helped me to keep my thinking straight when double-checking all the cases, including the case where I think I found a mistaken `return true` for the case of `trn <: #read`.